### ISSUE TITLE
Test photon inside P

### ISSUE THREAD

**Issue by matthewmueller** [prisma/prisma2#85]
Placeholder. Reminder to test if this works or not.

**Comment by divyenduz** [comment]
Doesn't look like it is a [use-case yet](https://github.com/cloudflare/wrangler/issues/312#issuecomment-510637738).

But maybe [in future](https://community.cloudflare.com/t/what-if-workers-could-access-your-sql-nosql-database/88805/6).

**Comment by Cerberus** [comment]
This is a favor thing that I looking for.

I have tried to implement and upload photonJS to cf-worker manually (without query engine binary). Separate the photonJS on worker & the engine on my server.

In Most cases, it works. But The challenge parts to deal are:

- Cloudflare workers limit startup-time **200 ms** for uploading script. I failed to upload on sometimes with 2 mini generated data-models and a little third party npm. We must to keep it minimal.

- Cloudflare workers limit 1 MB per script. After I install `apollo-server-cloudflare`, it takes ~450 kB. And if you prefer to use `nexus` (code-first implementation style) you have to set 2 schema files for development environment (that use nexus) and production environment (for uploading to worker). Because in nexus package, it contains a huge package like `prettier` that will make your script size over the quota.

I hope this [tier 2](https://github.com/prisma/specs/tree/master/binaries#tier-2) come soon. : )

**Comment by divyenduz** [comment]
@matthewmueller Assigned because you would leave a comment was the outcome of retriaging meeting 🙏

**Comment by matthewmueller** [comment]
Note: unassigning myself on things I'm not actively working on. I may pick this up in the future.

**Comment by IljaDaderko** [comment]
Not sure if this eases some concerns above, but they announced new upgrade "Workers Unbound" https://www.cloudflare.com/workers-unbound-beta

**Comment by Siilwyn** [comment]
For those following this issue, some days ago an upstream issue got an update about this:
https://github.com/cloudflare/db-connect/issues/10#issuecomment-785386626

TL;DR, at the moment TCP support is not yet supported by Cloudflare.

**Comment by janpio** [comment]
Not looking too good: https://github.com/cloudflare/db-connect is deprecated: https://github.com/cloudflare/db-connect/issues/10#issuecomment-785386626 The description makes it pretty clear, this is not a current priority 😭 

Unless someone builds a HTTP Data API for Prisma of course... but that only makes it work, not necessarily make it work great.

**Comment by cabello** [comment]
I tried following that [guide](https://www.prisma.io/docs/guides/deployment/deployment-guides/deploying-to-cloudflare-workers) but using PostgreSQL instead and I can't get it to work.

```
❯ npx wrangler dev
Error: Something went wrong with the request to Cloudflare...
Script startup timed out.
 [API code: 10021]
❯ npx wrangler publish
✨  Built successfully, built project size is 366 KiB.
Error: Something went wrong with the request to Cloudflare...
Script startup timed out.
 [API code: 10021]
```

At first I thought it was this issue https://github.com/cloudflare/wrangler/issues/1129 because:

```
❯ npx wrangler secret put DATABASE_URL
Enter the secret text you'd like assigned to the variable DATABASE_URL on the script named prisma-pg-cloudflare:
****
🌀  Creating the secret for script name prisma-pg-cloudflare
🌀  Worker prisma-pg-cloudflare doesn't exist in the API yet. Creating a draft Worker so we can create new secret.
Error: Something went wrong with the request to Cloudflare...
Script startup timed out.
 [API code: 10021]
```

Then I created the project using the web console and I was able to set the secret:

```
❯ npx wrangler secret put DATABASE_URL
Enter the secret text you'd like assigned to the variable DATABASE_URL on the script named prisma-pg-cloudflare:
****
🌀  Creating the secret for script name prisma-pg-cloudflare
✨  Success! Uploaded secret DATABASE_URL.
```

The project still fails to dev/publish.

I thought it could be because the project has 0.3MB but both free & plan have the same 1MB limits:
https://developers.cloudflare.com/workers/platform/limits#account-plan-limits

The `dist` and `worker` scripts are correctly generated, they just seen to hang when booting up which causes the time out, no idea what is causing the hang.

**Comment by matthewmueller** [comment]
Thanks for the write-up @cabello!

As @2color mentioned, we've written [a guide](https://www.prisma.io/docs/guides/deployment/deployment-guides/deploying-to-cloudflare-workers) on how to get this working. We also have a video presentation that we gave on [Cloudflare TV](https://cloudflare.tv/event/4Bd3dBeY7bKy7fACFPCRdk).

**Comment by kettanaito** [comment]
On `"@prisma/client": "^3.9.1"` it's no longer possible to follow the mentioned [guide](https://www.prisma.io/docs/guides/deployment/deployment-guides/deploying-to-cloudflare-workers). I'm getting the following error on `wranger dev`:

```sh
10021: Uncaught TypeError: Global addEventListener() cannot be used in modules. Instead, event handlers should be declared as exports on the root module.
  at line 0

    at throwFetchError (/my-project/node_modules/wrangler/wrangler-dist/cli.js:109771:17)
    at fetchResult (/my-project/node_modules/wrangler/wrangler-dist/cli.js:109742:5)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async previewToken (/my-project/node_modules/wrangler/wrangler-dist/cli.js:112163:29)
    at async createWorker (/my-project/node_modules/wrangler/wrangler-dist/cli.js:112180:17)
    at async start (/my-project/node_modules/wrangler/wrangler-dist/cli.js:113358:16) {
  code: 10021
}
```

This happens because importing `@prisma/client` in the worker source code somehow forces the entire bundle to be treated as a module:

```js
// node_modules/@miniflare/core/dist/src/index.js
  addEventListener = (type, listener, options) => {
    console.warn('debug:', this.#modules)

    if (this.#modules) {
      throw new TypeError("Global addEventListener() cannot be used in modules. Instead, event handlers should be declared as exports on the root module.");
    }
```

The `CodePlugin` from `@miniflare` gets instantiated with `modules: true`. Respectively, it throws a runtime exception stating you can't use worker global scope with modules (although nothing is marking my worker as a module). 

It'd be really great to have a working repo rather than a guideline page. I'd be thankful for any help with this.

**Comment by janpio** [comment]
Can you open this as a new bug issue @kettanaito? Then we can put this into our normal process and make sure this is fixed or properly documented.
